lib/block_view.go

```diff
@@ -4726,135 +5590,780 @@ func (bav *UtxoView) _connectUpdateProfile(
+	}
+
+	if totalBidderInput < txMeta.BidAmountNanos {
+		return 0, 0, nil, RuleErrorAcceptNFTBidderInputsInsufficientForBidAmount
+	}
+
+	// The bidder gets back any unspent nanos from the inputs specified.
+	bidderChangeNanos := totalBidderInput - txMeta.BidAmountNanos
+	// The amount of bitclout that should go to the original creator from this purchase.
+	// Calculated as: (BidAmountNanos * NFTRoyaltyToCreatorBasisPoints) / (100 * 100)
+	creatorRoyaltyNanos := IntDiv(
+		IntMul(
+			big.NewInt(int64(txMeta.BidAmountNanos)),
+			big.NewInt(int64(nftPostEntry.NFTRoyaltyToCreatorBasisPoints))),
+		big.NewInt(100*100)).Uint64()
+	// The amount of bitclout that should go to the original creator's coin from this purchase.
+	// Calculated as: (BidAmountNanos * NFTRoyaltyToCoinBasisPoints) / (100 * 100)
+	creatorCoinRoyaltyNanos := IntDiv(
+		IntMul(
+			big.NewInt(int64(txMeta.BidAmountNanos)),
+			big.NewInt(int64(nftPostEntry.NFTRoyaltyToCoinBasisPoints))),
+		big.NewInt(100*100)).Uint64()
+	fmt.Printf("Bid amount: %d, coin basis points: %d, coin royalty: %d",
+		txMeta.BidAmountNanos, nftPostEntry.NFTRoyaltyToCoinBasisPoints, creatorCoinRoyaltyNanos)
+	bidAmountMinusRoyalties := txMeta.BidAmountNanos - creatorRoyaltyNanos - creatorCoinRoyaltyNanos
+
+	// Connect basic txn to get the total input and the total output without
+	// considering the transaction metadata.
+	totalInput, totalOutput, utxoOpsForTxn, err := bav._connectBasicTransfer(
+		txn, txHash, blockHeight, verifySignatures)
+	if err != nil {
+		return 0, 0, nil, errors.Wrapf(err, "_connectAcceptNFTBid: ")
+	}
+
+	// Force the input to be non-zero so that we can prevent replay attacks.
+	if totalInput == 0 {
+		return 0, 0, nil, RuleErrorAcceptNFTBidRequiresNonZeroInput
+	}
+
+	if verifySignatures {
+		// _connectBasicTransfer has already checked that the transaction is
+		// signed by the top-level public key, which we take to be the poster's
+		// public key.
+	}
+
+	// Now we are ready to accept the bid. When we accept, the following must happen:
+	// 	(1) Update the nft entry with the new owner and set it as "not for sale".
+	//  (2) Delete all of the bids on this NFT since they are no longer relevant.
+	//  (3) Pay the seller.
+	//  (4) Pay royalties to the original creator.
+	//  (5) Pay change to the bidder.
+	//  (6) Add creator coin royalties to bitclout locked.
+	//  (7) Decrement the nftPostEntry NumNFTCopiesForSale.
+
+	// (1) Set an appropriate NFTEntry for the new owner.
+
+	newNFTEntry := &NFTEntry{
+		LastOwnerPKID:  updaterPKID.PKID,
+		OwnerPKID:      txMeta.BidderPKID,
+		NFTPostHash:    txMeta.NFTPostHash,
+		SerialNumber:   txMeta.SerialNumber,
+		IsForSale:      false,
+		UnlockableText: txMeta.UnlockableText,
+
+		LastAcceptedBidAmountNanos: txMeta.BidAmountNanos,
+	}
+	bav._setNFTEntryMappings(newNFTEntry)
+
+	// append the accepted bid entry to the list of accepted bid entries
+	prevAcceptedBidHistory := bav.GetAcceptNFTBidHistoryForNFTKey(&nftKey)
```

Comment:
Kinda feels like this should be a part of the NFTEntry but seems fine to leave it as is.

Reply:
We decided to keep NFT accepted bid history separate from the NFTEntry because it seemed annoying to have a potentially massive slice being gob encoded / decoded as part of the NFTEntry.  Not sure if this is a reasonable concern or not but it seemed like there was no harm in keeping it separate.